src/main/java/com/mineboundteam/minebound/magic/network/MagicSync.java

```diff
@@ -0,0 +1,116 @@
+package com.mineboundteam.minebound.magic.network;
+
+import java.util.function.Supplier;
+
+import com.mineboundteam.minebound.MineBound;
+import com.mineboundteam.minebound.capabilities.ArmorSpellsProvider;
+import com.mineboundteam.minebound.capabilities.PlayerSelectedSpellsProvider;
+import com.mineboundteam.minebound.capabilities.PlayerSelectedSpellsProvider.SelectedSpell;
+import com.mineboundteam.minebound.inventory.SelectSpellMenu;
+import com.mineboundteam.minebound.item.armor.MyrialArmorItem;
+
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.SimpleMenuProvider;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
+import net.minecraftforge.network.NetworkEvent;
+import net.minecraftforge.network.NetworkRegistry;
+import net.minecraftforge.network.simple.SimpleChannel;
+
+public class MagicSync {
+    public static final SimpleChannel NET_CHANNEL = NetworkRegistry.newSimpleChannel(
+            new ResourceLocation(MineBound.MOD_ID, "magic_sync"),
+            () -> "V1", a -> a.equals("V1"), a -> a.equals("V1"));
+
+    public static void registerPackets(FMLCommonSetupEvent event) {
+        NET_CHANNEL.registerMessage(0, ButtonMsg.class, ButtonMsg::encode, ButtonMsg::decode,
+                MagicSync::handleButtons);
+    }
+
+    public static void handleButtons(ButtonMsg msg, Supplier<NetworkEvent.Context> ctx) {
+        ctx.get().enqueueWork(() -> {
+            ServerPlayer player =  ctx.get().getSender();
+            if(!player.level.isClientSide()){
+                ItemStack chest = player.getItemBySlot(EquipmentSlot.CHEST);
```

Comment:
How come we are only checking the chestpiece here and no other armor?

Reply:
only the chestpiece has the hand slots on it; if you're not wearing a chestpiece then you won't have access to the spell library

Reply:
So even if the player has active spells equipped on other pieces, they won't be able to select and use them unless they have a Myrial chestpiece equipped?

Reply:
Yeah, I think we also have a bauble that gives that capability